Firestore/core/src/firebase/firestore/local/local_store.cc

```diff
@@ -377,17 +382,42 @@ bool LocalStore::ShouldPersistQueryData(const QueryData& new_query_data,
   return changes > 0;
 }
 
+absl::optional<QueryData> LocalStore::GetQueryData(const core::Query& query) {
+  auto target_id = target_id_by_query_.find(query);
+  if (target_id != target_id_by_query_.end()) {
+    return query_data_by_target_[target_id->second];
+  }
+  return query_cache_->GetTarget(query);
+}
+
 void LocalStore::NotifyLocalViewChanges(
     const std::vector<local::LocalViewChanges>& view_changes) {
   persistence_->Run("NotifyLocalViewChanges", [&] {
     for (const LocalViewChanges& view_change : view_changes) {
+      int target_id = view_change.target_id();
```

Comment:
Port of https://github.com/firebase/firebase-android-sdk/blob/d8ff6a54d12fd676a31c2c331f09fe0a7d033202/firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java#L481